src/adapters/sqlite/makeDispatcher/node/DatabaseDriver.js

```diff
@@ -0,0 +1,281 @@
+// @flow
+
+import Database from './Database'
+
+function fixArgs(args: any): any {
+  return Object.keys(args).reduce((acc, argName) => {
+    if (typeof acc[argName] === 'boolean') {
+      acc[argName] = acc[argName] ? 1 : 0
+    }
+    return acc
+  }, args)
+}
+
+type Migrations = { from: number, to: number, sql: string }
+
+class MigrationNeededError extends Error {
+  databaseVersion: number
+
+  type: string
+
+  constructor(databaseVersion): void {
+    super('MigrationNeededError')
+    this.databaseVersion = databaseVersion
+    this.type = 'MigrationNeededError'
+    this.message = 'MigrationNeededError'
+  }
+}
+
+class SchemaNeededError extends Error {
+  type: string
+
+  constructor(): void {
+    super('SchemaNeededError')
+    this.type = 'SchemaNeededError'
+    this.message = 'SchemaNeededError'
+  }
+}
+
+function getPath(dbName): string {
+  // If starts with `file:` or contains `/`, it's a path!
+  if (dbName === ':memory:' || dbName.indexOf('file:') === 0 || dbName.indexOf('/') >= 0) {
+    return dbName
+  }
+  return `${process.cwd()}/${dbName}`
+}
+
+class DatabaseDriver {
+  static sharedMemoryConnections = {}
+
+  database: Database
+
+  cachedRecords: any = {}
+
+  initialize = (dbName: string, schemaVersion: number) => {
+    this.init(dbName)
+    this.isCompatible(schemaVersion)
+  }
+
+  setUpWithSchema = (dbName: string, schema: string, schemaVersion: number) => {
+    this.init(dbName)
+    this.unsafeResetDatabase({ version: schemaVersion, sql: schema })
+    this.isCompatible(schemaVersion)
+  }
+
+  setUpWithMigrations = (dbName: string, migrations: Migrations) => {
+    this.init(dbName)
+    this.migrate(migrations)
+    this.isCompatible(migrations.to)
+  }
+
+  init = (dbName: string) => {
+    this.database = new Database(getPath(dbName))
+
+    const isSharedMemory = dbName.indexOf('mode=memory') > 0 && dbName.indexOf('cache=shared') > 0
+    if (isSharedMemory) {
+      if (!DatabaseDriver.sharedMemoryConnections[dbName]) {
+        DatabaseDriver.sharedMemoryConnections[dbName] = this.database
+      }
+      this.database = DatabaseDriver.sharedMemoryConnections[dbName]
+    }
+  }
+
+  find = (table: string, id: string) => {
+    if (this.isCached(table, id)) {
+      return id
+    }
+
+    const query = `SELECT * FROM ${table} WHERE id == ? LIMIT 1`
```

Comment:
check for the current version of database drivers on master. There's been some extra escaping added -- I actually thing this shouldn't pass tests, unless you're not branching off of master

Reply:
I'll rebase, just in case...

Reply:
I checked and it was missing on that one. Also, the tests were working OK on `master`.